src/coreclr/tools/Common/TypeSystem/IL/Stubs/PInvokeILEmitter.cs

```diff
@@ -380,6 +388,19 @@ private PInvokeILStubMethodIL EmitIL()
             if (_targetMethod.HasCustomAttribute("System.Runtime.InteropServices", "LCIDConversionAttribute"))
                 throw new NotSupportedException();
 
+            if (_callingConvention == UnmanagedCallingConventions.Swift)
+            {
+                // Swift calling convention has strict rules about value types that the JIT does not implement.
+                // Throw an exception if we encounter a value type that is not a primitive (and as such needs this lowering logic applied).
+                foreach (var marshaller in _marshallers)
+                {
+                    if (marshaller.ManagedType is { IsValueType: true, IsEnum: false, IsPrimitive: false })
```

Comment:
Would it handle swift types as special cases and allow them in the signature?

Reply:
Yes, I need to update this so SwiftSelf is allowed (SwiftError is passed by pointer, so it doesn't need to be special-cased in this check)